### PR TITLE
feat(movie-web): add presence

### PR DESCRIPTION
**Pull request by Jordaar**
## Description 

Added [movie-web.app](https://movie-web.app/), a small web app for watching your favourite movies and TV shows with no ads or popups.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/M/movie-web/metadata.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		"name": "jordaar"
+	},
+	"service": "movie-web",
+	"description": {
+		"en": "The place for your favorite movies & shows"
+	},
+	"url": "movie-web.app",
+	"regExp": "(dev.)?(movie-web.app)",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/VyageyF.png",
```
Comment (plskz) [inline websites/M/movie-web/metadata.json:14]: Logo must be `512x512`

```suggestion
	"logo": "https://i.imgur.com/EW8lOTs.png",
```

File: websites/M/movie-web/metadata.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"author": {
+		"id": "705490775932403795",
+		"name": "jordaar"
+	},
+	"service": "movie-web",
+	"description": {
+		"en": "The place for your favorite movies & shows"
+	},
+	"url": "movie-web.app",
+	"regExp": "(dev.)?(movie-web.app)",
```
Comment (DarkVillager) [inline websites/M/movie-web/metadata.json:12]:
```suggestion
	"regExp": "(dev[.])?(movie-web.app)",
```

File: websites/M/movie-web/presence.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+		seasonId: string;
+	};
+	progress: {
+		time: number;
+		duration: number;
+	};
+}
+
+presence.on("UpdateData", async () => {
+	const { pathname, href, host } = window.location,
```
Comment (DarkVillager) [inline websites/M/movie-web/presence.ts:135]: We prefer documen.location
```suggestion
	const { pathname, href, host } = document.location,
```

File: websites/M/movie-web/presence.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+					now.getSeconds() + progress.duration - progress.time
+				);
+			}
+		}
+	}
+
+	if (showWatchButton && !pathname.startsWith("/search")) {
+		presenceData.buttons = [
+			{
+				label: "Watch",
```
Comment (DarkVillager) [inline websites/M/movie-web/presence.ts:261]:
```suggestion
				label: "Watch Video",
```

File: websites/M/movie-web/presence.ts [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+			} else presenceData.details = title;
+
+			if (state.mediaPlaying.isFirstLoading) {
+				presenceData.smallImageKey = "https://i.imgur.com/lBCZWIB.gif";
+				presenceData.smallImageText = "Loading";
+			} else if (state.mediaPlaying.isPlaying) {
+				const now = new Date();
+				presenceData.endTimestamp = now.setSeconds(
+					now.getSeconds() + state.progress.duration - state.progress.time
+				);
```
Comment (DarkVillager) [inline websites/M/movie-web/presence.ts:200]: Why not use the normal way to get timestamps? (Lmk if this is due to api usage or sum)
image

```suggestion
				[, presenceData.endTimestamp] = presence.getTimestampsfromMedia(document.querySelector("video"))
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/movie-web/presence.ts (hunk 5 above)
Comment (Jordaar) [reply websites/M/movie-web/presence.ts:200]: We expose time and duration in metadata anyway. Can't we go with it?

File: websites/M/movie-web/presence.ts (hunk 4 above)
Comment (Jordaar) [reply websites/M/movie-web/presence.ts:261]: Can't we keep it to just "Watch"?

Comment (DarkVillager) [reply websites/M/movie-web/presence.ts:261]: > Can't we keep it to just "Watch"?

We typically use Watch Video if its about a video

File: websites/M/movie-web/presence.ts (hunk 5 above)
Comment (DarkVillager) [reply websites/M/movie-web/presence.ts:200]: > We expose time and duration in metadata anyway. Can't we go with it?

As far as i know this is prefferd

File: websites/M/movie-web/metadata.json (hunk 2 above)
Comment (Jordaar) [reply websites/M/movie-web/metadata.json:12]: The suggested regex is not working with dev.movie-web.app

Comment (theusaf) [reply websites/M/movie-web/metadata.json:12]: Strange, the regex should match that domain string. I've tested it in the javascript console/regex websites.
Also, for consistency:
```suggestion
	"regExp": "(dev[.])?(movie-web[.]app)",
```

File: websites/M/movie-web/presence.ts (hunk 4 above)
Comment (Bas950) [reply websites/M/movie-web/presence.ts:259]: You can probs do something like:
```ts
`Watch ${meta.meta.type === "series" ? "Series" : "Video"}`
```

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**